lib/ansiblelint/formatters/__init__.py

```diff
@@ -87,6 +87,33 @@ def format(self, match, colored=False):
                                     match.message)
 
 
+class AnnotationsFormatter(BaseFormatter):
+    # https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-a-warning-message
+    """Emits errors in format that is recognized by Github Annotation parsing.
+
+    ::warning file={name},line={line},col={col}::{message}
+
+    Supported levels: debug, warning, error
+    """
+
+    def format(self, match, colored=False):
+        formatstr = u"::{0} file={1},line={2}::[{3}] {4}"
+        return formatstr.format(
+            self.severity_to_level(match.rule.severity),
+            self._format_path(match.filename),
+            match.linenumber,
+            "E" + match.rule.id,
+            match.message)
+
+    def severity_to_level(self, severity: str) -> str:
```

Comment:
I think this shouldn't be embedded into the class (it does nothing with the instance. But if you want to keep it, turn into a `@staticmethod` at least, and add a leading underscore.